src/main.cpp

```diff
@@ -403,39 +403,287 @@ bool FullSyncWithDPORNodes()
+    // small, we are mounting to high CDFs. This means to be reasonably accurate, we cannot just add the probabilities, because the intersections
+    // become significant. The CDF of a compound geometric distribution as you do tosses with different probabilities follows the
+    // recursion relation... CDF.i = 1 - (1 - CDF.i-1)(1 - p.i). If all probabilities are the same, this reduces to the familiar
+    // CDF.k = 1 - (1 - p)^k where ^ is exponentiation.
+    for(const auto& itertime : UniqueUTXOTimes)
+    {
+        
+        nTime = itertime;
+        dProbAccumulator = 0;
+        
+        for( auto& iterUTXO : vUTXO)
+        {
+        
+            if (fDebug10) LogPrintf("ETTS debug: Unique UTXO Time: %u, vector pair <%u, %u>", nTime, iterUTXO.first, iterUTXO.second);
+             
+            // If the "negative Gantt chart bar" is ending or has ended for a UTXO, it now accumulates probability. (I.e. the event time being checked
+            // is greater than or equal to the cooldown expiration of the UTXO.)
+            // accumulation for that UTXO.
+            if(nTime >= iterUTXO.first)
+            {
+                // The below weight calculation is just like the CalculateStakeWeightV8 in kernel.cpp.
+                nCoinWeight = iterUTXO.second / 1250000;
+                
+                dProbAccumulator = 1 - ((1 - dProbAccumulator) * (1 - (dUnitStakeProbability * nCoinWeight)));
+                if (fDebug10) LogPrintf("ETTS debug: dProbAccumulator = %e", dProbAccumulator);
+            }
+
+        }
+        nDeltaTime = nTime - nTimePrev;
+        nThrows = nDeltaTime / (STAKE_TIMESTAMP_MASK + 1);
+        if (fDebug10) LogPrintf("ETTS debug: nThrows = %i", nThrows);
+        dCumulativeProbability = 1 - ((1 - dCumulativeProbability) * pow((1 - dProbAccumulator), nThrows));
+        if (fDebug10) LogPrintf("ETTS debug: dCumulativeProbability = %e", dCumulativeProbability);
+        
+        if(dCumulativeProbability >= dConfidence)
+            break;
+        
+        nTimePrev = nTime;
+    }
+
+    // If (dConfidence - dCumulativeProbability) > 0, it means we exited the negative Gantt chart area and the desired confidence level
+    // has not been reached. All of the eligible UTXO's are contributing probability, and this is the final dProbAccumulator value.
+    // If the loop above is degenerate (i.e. only the current time pass through), then dCumulativeProbability will be zero.
+    // If it was not degenerate and the positive reqions in the Gantt chart area contributed some probability, then dCumulativeProbability will
+    // be greater than zero. We must compute the amount of time beyond nTime that is required to bridge the gap between
+    // dCumulativeProbability and dConfidence. If (dConfidence - dCumulativeProbability) <= 0 then we overshot during the Gantt chart area,
+    // and we will back off by nThrows amount, which will now be negative.
```

Comment:
Yeah... it is a bit wild, but better than looping every ETTS mask quantum to prevent overshooting. That would require 225+ loops. Most of the time there will only be a few UTXO ends “events” in the outer loop... in many instances only 1 to 3, and this reduces the loop workload both outer and inner considerably...